oxy_fitting_ver2.py

```diff
@@ -0,0 +1,447 @@
+
+# coding: utf-8
+
+# # THIS IS ANALYSIS OF 1707 (PO6E) based upon NOAA's oxy fitting
+# 
+# ## Density Matching Instead of Interpolation
+
+# In[1]:
+
+
+import sys
+sys.path.append('/Users/k3jackson/p06e/ctd_proc')
+sys.path.append('/Users/k3jackson/p06e/Kennycode/')
+import matplotlib
+matplotlib.use('TkAgg')
+import matplotlib.pyplot as plt
+import scipy
+import numpy as np
+import libODF_process_ctd as process_ctd
+import libODF_sbe_reader as sbe_rd
+import libODF_sbe_equations_dict as sbe_eq
+import gsw
+import pandas as pd
+
+
+
+## In[2]:
+#
+#
+#raw_dir = '/Users/k3jackson/NPD1707/raw/'
+#ssscc_file = '/Users/k3jackson/NPD1707/ssscc.csv'
+#time_dir = '/Users/k3jackson/NPD1707/time/'
+#btl_dir = '/Users/k3jackson/NPD1707/bottle/'
+#
+#sal_col='CTDSAL'
+#t_col='CTDTMP1'
+#p_col='CTDPRS'
+#lon_col='GPSLON'
+#lat_col='GPSLAT'
+#sal_btl_col='CTDSAL'
+#t_btl_col='CTDTMP1'
+#p_btl_col='CTDPRS'
+#dov_col = 'CTDOXYVOLTS'
+#lat_btl_col='GPSLAT'
+#lon_btl_col='GPSLON'
+#oxy_btl_col='CTDOXY1'
+#dov_btl_col='CTDOXYVOLTS'
+#
+#method = 3
+#
+#dataframe_concat = pd.DataFrame()
+#ssscc = []
+#with open(ssscc_file, 'r') as filename:
+#    ssscc = [line.strip() for line in filename]
+#for cast in range(len(ssscc)):
+#    stn_nbr = int(ssscc[cast][0:3])
+#    cst_nbr = int(ssscc[cast][-2:])
+#    stn_cst = str(ssscc[cast])
+#
+#    hexfile = raw_dir+stn_cst+'.hex'
+#    xmlfile = raw_dir+stn_cst+'.XMLCON'
+#    
+#    time_file = time_dir+stn_cst+'_time.pkl'
+#    btl_file = btl_dir+stn_cst+'_btl_mean.pkl'
+#    
+#    time_data = process_ctd.dataToNDarray(time_file,float,True,',',1)
+#    time_data = pd.DataFrame.from_records(time_data)  
+#    
+#    btl_data = process_ctd.dataToNDarray(btl_file,float,True,',',0) 
+#    btl_data = pd.DataFrame.from_records(btl_data)
+#
+
+def oxy_fit(time_data,btl_data,ssscc,hexfile,xmlfile,method=3,sal_col='CTDSAL',\
+            t_col='CTDTMP1',p_col='CTDPRS',lon_col='GPSLON',lat_col='GPSLAT',\
+            dov_col='CTDOXYVOLTS',sal_btl_col='CTDSAL',t_btl_col='CTDTMP1',\
+            p_btl_col='CTDPRS',lat_btl_col='GPSLAT',lon_btl_col='GPSLON',\
+            dov_btl_col='CTDOXYVOLTS',oxy_btl_col='CTDOXY1'):
+    
+    ##NORMALLY INT#####
+    stn_nbr = int(ssscc[0:3])
+    cst_nbr = int(ssscc[-2:])
+#    stn_cst = str(ssscc)
+    
+    time_data['CT_time'] = gsw.CT_from_t(time_data[sal_col],time_data[t_col],time_data[p_col])
```

Comment:
For logical blocks of actions, please write a comment with a quick explanation of what's going on here. We possibly want to convert these logical blocks to methods, to make the function more readable, atomic (properly scoped) and make sure we aren't leaking out across the main function.